python/ray/serve/master.py

```diff
@@ -424,7 +426,24 @@ def _scale_replicas(self, backend_tag, num_replicas):
         current_num_replicas = len(self.replicas[backend_tag])
         delta_num_replicas = num_replicas - current_num_replicas
 
+        _, _, replica_config = self.backends[backend_tag]
         if delta_num_replicas > 0:
+            can_schedule = try_schedule_resources_on_nodes(
+                requirements=[
+                    replica_config.resource_dict
+                    for _ in range(delta_num_replicas)
+                ],
+                ray_nodes=ray.nodes())
+            if not all(can_schedule):
+                num_possible = sum(can_schedule)
+                raise RayServeException(
+                    "Cannot scale backend {} to {} replicas. Ray Serve tried "
+                    "to add {} replicas but the resources only allows {} "
+                    "to be added. To fix this, consider scaling to replica to "
```

Comment:
```suggestion
                    "to be added. To fix this, consider scaling to replica to "
```

I would also add "... or add more resources to the cluster. You can check available resources with ray.nodes()."